### PR TITLE
[C3] pnpm fixes

### LINKED ISSUE #3310: 🐛 BUG: `create-cloudflare` (C3) with pnpm throws error.
### Which Cloudflare product(s) does this pertain to?

C3

### What version of `Wrangler` are you using?

\-

### What operating system are you using?

Pop_OS 22.04 (Ubuntu)

### Describe the Bug

Creating new project with `create-cloudflare` with `pnpm` throws error. using `npm` working properly.
node v18.16.0
pnpm v8.5.1
```shell
$ pnpm create cloudflare
.../share/pnpm/store/v3/tmp/dlx-16286    |   +1 +
Packages are hard linked from the content-addressable store to the virtual store.
  Content-addressable store is at: /home/devin/.local/share/pnpm/store/v3
  Virtual store is at:             ../.local/share/pnpm/store/v3/tmp/dlx-16286/node_modules/.pnpm
.../share/pnpm/store/v3/tmp/dlx-16286    | Progress: resolved 1, reused 1, downloaded 0, added 1, done

using create-cloudflare version 2.0.7

╭ Create an application with Cloudflare Step 1 of 3
│ 
├ Where do you want to create your application?
│ dir quiet-wood-a6ad
│
├ What type of application do you want to create?
│ type Website or web app
│
├ Which development framework do you want to use?
│ framework Next
│
╰ Continue with Next via `pnpx create-next-app@13.3.4 quiet-wood-a6ad`

node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: spawn pnpx ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn pnpx',
  path: 'pnpx',
  spawnargs: [ 'create-next-app@13.3.4', 'quiet-wood-a6ad' ]
}

Node.js v18.16.0
 ERROR  Command failed with exit code 1: create-cloudflare

pnpm: Command failed with exit code 1: create-cloudflare
    at makeError (/snapshot/dist/pnpm.cjs)
    at handlePromise (/snapshot/dist/pnpm.cjs)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.handler (/snapshot/dist/pnpm.cjs)
    at async /snapshot/dist/pnpm.cjs
    at async main (/snapshot/dist/pnpm.cjs)
    at async runPnpm (/snapshot/dist/pnpm.cjs)
    at async /snapshot/dist/pnpm.cjs
```

### PR DESCRIPTION
**Pull request by jculvey**
Fixes #3310.

**What this PR solves / how to test:**

Fixes usage via pnpm 6+ where `pnpx` has been deprecated in favor of `pnpm dlx`. Makes a similar adjustment for yarn.

**Author has included the following, where applicable:**

- [ x ] Tests
- [ x ] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

### CODE REVIEW

File: packages/create-cloudflare/src/frameworks/angular/index.ts [hunk 1]
```diff
@@ -3,24 +3,24 @@ import { resolve } from "node:path";
...
 	runFrameworkGenerator,
 } from "helpers/command";
 import { readFile, readJSON, writeFile } from "helpers/files";
 import { spinner } from "helpers/interactive";
+import { detectPackageManager } from "helpers/packages";
 import { getFrameworkVersion } from "../index";
 import type { PagesGeneratorContext, FrameworkConfig } from "types";
 
-const { npx, npm } = detectPackageManager();
+const { dlx, npx, npm } = detectPackageManager();
```
Comment (petebacondarwin) [inline packages/create-cloudflare/src/frameworks/angular/index.ts:16]: Why is `npx` still here? Why are only some of the `npx`s changed to `dlx`?

Comment (jculvey) [reply packages/create-cloudflare/src/frameworks/angular/index.ts:16]: Yeah, specifically just the bits where we invoke the framework scaffolding CLIs. `dlx` is designed for those situations where you aren't in a package context and don't want to invoke something from node-modules

**Review by RamIdeas — APPROVED** [review]

**Review by JacobMGEvans — APPROVED** [review]

File: packages/create-cloudflare/src/frameworks/angular/index.ts [hunk 2]
```diff
@@ -44,10 +44,10 @@ const config: FrameworkConfig = {
 	packageScripts: {
 		process:
 			"node ./tools/copy-worker-files.mjs && node ./tools/copy-client-files.mjs && node ./tools/bundle.mjs",
-		prestart: "npm run build:ssr && npm run process",
+		prestart: `${npm} run build:ssr && npm run process`,
```
Comment (petebacondarwin) [inline packages/create-cloudflare/src/frameworks/angular/index.ts:47]: Oops!

Should be:

```
`${npm} run build:ssr && ${npm} run process`
```

Comment (petebacondarwin) [reply packages/create-cloudflare/src/frameworks/angular/index.ts:47]: Fixed in https://github.com/cloudflare/workers-sdk/pull/3486/files#diff-eafbea9c833fea865a227b2b86c2948493ac2c97f2c6f275572af5976c3628b9R47

**Pull request merged**